lang/py/avro/io.py

```diff
@@ -155,7 +155,7 @@ def validate(expected_schema, datum, raise_on_error=False):
         else:
             # the current node was not valid.
             if raise_on_error:
-                raise avro.errors.AvroTypeException(current_node.schema, current_node.datum)
+                raise avro.errors.AvroTypeException(current_node.schema, current_node.name, current_node.datum)
```

Comment:
Will this be a breaking change? Even if others use AvroTypeException directly, this `init` API should be internal.